File: peek_plugin_search/_private/storage/SearchPropertyTuple.py
from sqlalchemy import Column
from sqlalchemy import Integer, String
from vortex.Tuple import Tuple, addTupleType

from peek_plugin_search._private.PluginNames import searchTuplePrefix
from peek_plugin_search._private.storage.DeclarativeBase import DeclarativeBase


@addTupleType
class SearchPropertyTuple(Tuple, DeclarativeBase):
    __tupleType__ = searchTuplePrefix + 'SearchPropertyTuple'
    __tablename__ = 'SearchPropertyTuple'

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String, nullable=False)
    title = Column(String, nullable=False)

File: peek_plugin_search/plugin-module/_private/tuples/SearchPropertyTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {searchTuplePrefix} from "../PluginNames";


@addTupleType
export class SearchPropertyTuple extends Tuple {
    public static readonly tupleName = searchTuplePrefix + "SearchPropertyTuple";

    //  The name of the search property
    name: string;

    //  The title of the search property
    title: string;

    constructor() {
        super(SearchPropertyTuple.tupleName)
    }
}